sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_download_async.py

```diff
@@ -4,80 +4,93 @@
 # --------------------------------------------------------------------------
 # pylint: disable=invalid-overridden-method
+# mypy: disable-error-code=override
+
 import asyncio
 import sys
 import warnings
 from io import BytesIO
 from itertools import islice
-from typing import AsyncIterator
+from typing import (
+    Any, AsyncIterator, Awaitable, Generator,
+    Callable, cast, IO, Optional, Tuple, Union,
+    TYPE_CHECKING
+)
 
 from azure.core.exceptions import HttpResponseError, ResourceModifiedError
+
 from .._download import _ChunkDownloader
 from .._shared.request_handlers import validate_and_format_range_headers
-from .._shared.response_handlers import process_storage_error, parse_length_from_content_range
+from .._shared.response_handlers import parse_length_from_content_range, process_storage_error
+
+if TYPE_CHECKING:
+    from .._generated.aio.operations import FileOperations
+    from .._models import FileProperties
+    from .._shared.models import StorageConfiguration
 
 
-async def process_content(data):
+async def process_content(data: Any) -> bytes:
     if data is None:
         raise ValueError("Response cannot be None.")
 
     try:
-        return data.response.body()
+        return cast(bytes, data.response.body())
     except Exception as error:
         raise HttpResponseError(message="Download stream interrupted.", response=data.response, error=error) from error
 
 
 class _AsyncChunkDownloader(_ChunkDownloader):
-    def __init__(self, **kwargs):
+    def __init__(self, **kwargs: Any) -> None:
         super(_AsyncChunkDownloader, self).__init__(**kwargs)
-        self.stream_lock = asyncio.Lock() if kwargs.get('parallel') else None
-        self.progress_lock = asyncio.Lock() if kwargs.get('parallel') else None
+        self.stream_lock_async = asyncio.Lock() if kwargs.get('parallel') else None
+        self.progress_lock_async = asyncio.Lock() if kwargs.get('parallel') else None
 
-    async def process_chunk(self, chunk_start):
+    async def process_chunk(self, chunk_start: int) -> None:
         chunk_start, chunk_end = self._calculate_range(chunk_start)
         chunk_data = await self._download_chunk(chunk_start, chunk_end - 1)
         length = chunk_end - chunk_start
         if length > 0:
             await self._write_to_stream(chunk_data, chunk_start)
             await self._update_progress(length)
 
-    async def yield_chunk(self, chunk_start):
+    async def yield_chunk(self, chunk_start: int) -> bytes:
         chunk_start, chunk_end = self._calculate_range(chunk_start)
         return await self._download_chunk(chunk_start, chunk_end - 1)
 
-    async def _update_progress(self, length):
-        if self.progress_lock:
-            async with self.progress_lock:  # pylint: disable=not-async-context-manager
+    async def _update_progress(self, length: int) -> None:
+        if self.progress_lock_async:
+            async with self.progress_lock_async:  # pylint: disable=not-async-context-manager
```

Comment:
I _think_ we can remove this PyLint disable now that we renamed the lock variable.

Reply:
That did the magic 🧙